app/base/models.py

```diff
@@ -143,32 +145,25 @@ def get_property(cls, _prop_id):
 
     @classmethod
     def add_property(
-        cls, _name, _desc, _price, _location, _image_folder, _photos, _user_id
+        cls, prop_data
     ):
-        new_property = cls(
-            name=_name,
-            desc=_desc,
-            price=_price,
-            location=_location,
-            image_folder=_image_folder,
-            photos=_photos,
-            user_id=_user_id,
-        )
+        new_property = cls(**prop_data)
         db.session.add(new_property)
         db.session.commit()
 
     @classmethod
-    def update_property(
-        cls, prop_id, _name, _desc, _price, _location, _image_folder, _photos, _user_id
-    ):
-        prop_ro_update = cls.query.get(prop_id)
-        prop_ro_update.name = (_name,)
-        prop_ro_update.desc = (_desc,)
-        prop_ro_update.price = (_price,)
-        prop_ro_update.location = (_location,)
-        prop_ro_update.image_folder = (_image_folder,)
-        prop_ro_update.photos = (_photos,)
-        prop_ro_update.user_id = _user_id
+    def update_property(cls, prop_data, prop_id):
+        property_to_update = Property.query.filter_by(id=prop_id)
+        for key, value in prop_data.items():
+            property_to_update.update({key: value})
+            db.session.commit()
+
+    @classmethod
+    def update_property_images(cls, image_dir, img_list, prop_id):
+        prop_to_update = Property.query.get(prop_id)
+        shutil.rmtree(f"{current_app.root_path}/base/static/{prop_to_update.image_folder}")
+        prop_to_update.image_folder = image_dir  # deletes old property image folder including contents
+        prop_to_update.photos = img_list
         db.session.commit()
```

Comment:
this would be a good utility method for a celery worker
like it would be cool to have a worker utilizing this cleanup method running maybe every 30 minutes

Reply:
I will implement it.